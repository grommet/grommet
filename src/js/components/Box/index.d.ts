import * as React from 'react';
import {
  A11yTitleType,
  AlignType,
  AlignContentType,
  AlignSelfType,
  BackgroundType,
  BasisType,
  BorderType,
  DirectionType,
  ElevationType,
  FillType,
  GapType,
  GridAreaType,
  HeightType,
  MarginType,
  PadType,
  PolymorphicType,
  RoundType,
  WidthType,
  AnimationType,
  SkeletonType,
} from '../../utils';

export interface BoxProps {
  a11yTitle?: A11yTitleType;
  alignSelf?: AlignSelfType;
  gridArea?: GridAreaType;
  margin?: MarginType;
  align?: AlignType;
  alignContent?: AlignContentType;
  animation?: AnimationType;
  background?: BackgroundType;
  basis?: BasisType;
  border?: BorderType;
  cssGap?: boolean;
  direction?: DirectionType;
  elevation?: ElevationType;
  flex?: 'grow' | 'shrink' | boolean | { grow?: number; shrink?: number };
  fill?: FillType;
  focusIndicator?: boolean;
  gap?: GapType | { row?: GapType; column?: GapType };
  height?: HeightType;
  hoverIndicator?:
    | { background?: BackgroundType; elevation?: ElevationType }
    | BackgroundType
    | boolean;
  justify?:
    | 'around'
    | 'between'
    | 'center'
    | 'end'
    | 'evenly'
    | 'start'
    | 'stretch';
  overflow?:
    | 'auto'
    | 'hidden'
    | 'scroll'
    | 'visible'
    | {
        horizontal?: 'auto' | 'hidden' | 'scroll' | 'visible';
        vertical?: 'auto' | 'hidden' | 'scroll' | 'visible';
      }
    | string;
  pad?: PadType;
  responsive?: boolean | 'container';
  round?: RoundType;
  skeleton?: SkeletonType;
  tag?: PolymorphicType;
  as?: PolymorphicType;
  width?: WidthType;
  wrap?: boolean | 'reverse';
}

export interface BoxExtendedProps
  extends BoxProps,
    Omit<JSX.IntrinsicElements['div'], keyof BoxProps> {}

// Keep type alias for backwards compatibility.
export type BoxTypes = BoxProps & JSX.IntrinsicElements['div'];

declare const Box: React.FC<BoxExtendedProps>;

export { Box };
