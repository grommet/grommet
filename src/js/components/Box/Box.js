import React, {
  Children,
  forwardRef,
  useContext,
  useEffect,
  useMemo,
  useState,
} from 'react';

import { ThemeContext } from 'styled-components';
import {
  backgroundIsDark,
  useForwardedRef,
  deviceResponsive,
  getBreakpoint,
} from '../../utils';
import { Keyboard } from '../Keyboard';

import { StyledBox, StyledBoxGap } from './StyledBox';
import { BoxPropTypes } from './propTypes';
import { SkeletonContext, useSkeleton } from '../Skeleton';
import { AnnounceContext } from '../../contexts/AnnounceContext';
import { OptionsContext } from '../../contexts/OptionsContext';
import { ResponsiveContainerContext, ResponsiveContext } from '../../contexts';
import { useThemeValue } from '../../utils/useThemeValue';
import { supportsContainerQueries } from '../../utils/responsive';

// The value of ResponsiveContainerContext
// when we're within a responsive container
const responsiveContainerValue = true;

const Box = forwardRef(
  (
    {
      a11yTitle,
      background: backgroundProp,
      border,
      children,
      cssGap,
      direction = 'column',
      elevation, // munged to avoid styled-components putting it in the DOM
      fill, // munged to avoid styled-components putting it in the DOM
      gap,
      kind, // munged to avoid styled-components putting it in the DOM
      onBlur,
      onClick,
      onFocus,
      overflow, // munged to avoid styled-components putting it in the DOM
      responsive: responsiveProp = true,
      tag,
      as,
      wrap, // munged to avoid styled-components putting it in the DOM,
      width, // munged to avoid styled-components putting it in the DOM
      height, // munged to avoid styled-components putting it in the DOM
      tabIndex,
      skeleton: skeletonProp,
      ...rest
    },
    ref,
  ) => {
    const { theme, passThemeFlag } = useThemeValue();
    // boxOptions was created to preserve backwards compatibility but
    // should not be supported in v3
    const { box: boxOptions } = useContext(OptionsContext);

    console.log('as', as, typeof as);
    const skeleton = useSkeleton();

    const responsiveContainer = useContext(ResponsiveContainerContext);
    const responsive =
      responsiveContainer && responsiveProp ? 'container' : responsiveProp;

    let background = backgroundProp;

    const announce = useContext(AnnounceContext);

    const [containerBreakpoint, setContainerBreakpoint] = useState(
      deviceResponsive(navigator.userAgent, theme) ||
        theme?.global?.deviceBreakpoints.tablet,
    );

    const containerRef = useForwardedRef(ref);
    // console.log('render', responsiveProp, containerRef);

    // Update the size in the ResponsiveContext at this
    // level if we're in responsive container mode.
    useEffect(() => {
      // console.log('usEffect', responsiveProp, containerRef);
      const containerElement = containerRef?.current;
      if (responsiveProp !== 'container' || !containerElement) return undefined;

      let resizeObserver;
      if (typeof window !== 'undefined' && window.ResizeObserver) {
        resizeObserver = new window.ResizeObserver((entries) => {
          window.requestAnimationFrame(() => {
            if (!Array.isArray(entries) || !entries.length) {
              return;
            }
            const size = entries[0]?.borderBoxSize?.[0]?.inlineSize;
            // console.log('Observer size', size, getBreakpoint(size, theme));
            if (size) {
              setContainerBreakpoint(getBreakpoint(size, theme));
            }
          });
        });

        resizeObserver.observe(containerElement);
      }
      // Initial size for ResponsiveContext in responsive container mode.
      // Also is a fallback for server side rendering
      const containerWidth = containerElement.getBoundingClientRect().width;
      setContainerBreakpoint(getBreakpoint(containerWidth, theme));

      return () => {
        if (resizeObserver) {
          resizeObserver.disconnect();
        }
      };
    }, [theme, responsiveProp, containerRef]);

    useEffect(() => {
      if (skeletonProp?.message?.start) announce(skeletonProp.message.start);
      else if (typeof skeletonProp?.message === 'string')
        announce(skeletonProp.message);
      return () =>
        skeletonProp?.message?.end && announce(skeletonProp.message.end);
    }, [announce, skeletonProp]);

    const focusable = useMemo(
      () => onClick && !(tabIndex < 0),
      [onClick, tabIndex],
    );

    const [focus, setFocus] = useState();

    const clickProps = useMemo(() => {
      if (focusable) {
        return {
          onClick,
          onFocus: (event) => {
            setFocus(true);
            if (onFocus) onFocus(event);
          },
          onBlur: (event) => {
            setFocus(false);
            if (onBlur) onBlur(event);
          },
        };
      }
      const result = {};
      if (onBlur) result.onBlur = onBlur;
      if (onClick) result.onClick = onClick;
      if (onFocus) result.onFocus = onFocus;
      return result;
    }, [focusable, onClick, onFocus, onBlur]);

    const adjustedTabIndex = useMemo(() => {
      if (tabIndex !== undefined) return tabIndex;
      if (focusable) return 0;
      return undefined;
    }, [focusable, tabIndex]);

    if (
      (border === 'between' ||
        (border && border.side === 'between') ||
        (Array.isArray(border) && border.find((b) => b.side === 'between'))) &&
      !gap
    ) {
      console.warn('Box must have a gap to use border between');
    }

    let contents = children;
    if (
      gap &&
      gap !== 'none' &&
      (!(boxOptions?.cssGap || cssGap || typeof gap === 'object') ||
        // need this approach to show border between
        border === 'between' ||
        border?.side === 'between' ||
        (Array.isArray(border) && border.find((b) => b.side === 'between')))
    ) {
      // if border is an array, we need to extract the border between object
      const styledBoxGapBorder = Array.isArray(border)
        ? border.find((b) => b.side === 'between')
        : border;
      const boxAs = !as && tag ? tag : as;
      contents = [];
      let firstIndex;
      Children.forEach(children, (child, index) => {
        if (child) {
          if (firstIndex === undefined) {
            firstIndex = index;
          } else {
            contents.push(
              <StyledBoxGap
                // eslint-disable-next-line react/no-array-index-key
                key={`gap-${index}`}
                as={boxAs === 'span' ? boxAs : 'div'}
                gap={gap}
                directionProp={direction}
                responsive={responsive}
                border={styledBoxGapBorder}
                {...passThemeFlag}
              />,
            );
          }
        }
        contents.push(child);
      });
    }

    const nextSkeleton = useMemo(() => {
      // Decide if we need to add a new SkeletonContext. We need one if:
      //   1. skeleton info was set in a property OR
      //   2. there already is a SkeletonContext but this box has a
      //      background or border. This means the box probably is more
      //      distinguishable from the area around it.
      // We keep track of a depth so we know how to alternate backgrounds.
      if (skeletonProp || ((background || border) && skeleton)) {
        const depth = skeleton ? skeleton.depth + 1 : 0;
        return {
          ...skeleton,
          depth,
          ...(typeof skeletonProp === 'object' ? skeletonProp : {}),
        };
      }
      return undefined;
    }, [background, border, skeleton, skeletonProp]);

    let skeletonProps = {};
    if (nextSkeleton) {
      const {
        colors: skeletonThemeColors,
        size: skeletonThemeSize,
        ...skeletonThemeProps
      } = theme.skeleton;
      const skeletonColors = nextSkeleton.colors
        ? nextSkeleton.colors[theme.dark ? 'dark' : 'light']
        : skeletonThemeColors?.[theme.dark ? 'dark' : 'light'];
      skeletonProps = { ...skeletonThemeProps };
      background = skeletonColors[nextSkeleton.depth % skeletonColors.length];
      if (skeletonProp?.animation) {
        skeletonProps.animation = skeletonProp.animation;
      }
      contents = (
        <SkeletonContext.Provider value={nextSkeleton}>
          {contents}
        </SkeletonContext.Provider>
      );
    }

    // construct a new theme object in case we have a background that wants
    // to change the background color context
    const nextTheme = useMemo(() => {
      let result;
      if (background || theme.darkChanged) {
        const dark = backgroundIsDark(background, theme);
        const darkChanged = dark !== undefined && dark !== theme.dark;
        if (darkChanged || theme.darkChanged) {
          result = { ...theme };
          result.dark = dark === undefined ? theme.dark : dark;
          result.background = background;
        } else if (background) {
          // This allows DataTable to intelligently set the background
          // of a pinned header or footer.
          result = { ...theme };
          result.background = background;
        }
      }
      return result || theme;
    }, [background, theme]);

    let content = (
      <StyledBox
        as={!as && tag ? tag : as}
        aria-label={a11yTitle}
        background={background}
        border={border}
        ref={containerRef}
        directionProp={direction}
        elevationProp={elevation}
        fillProp={fill}
        focus={focus}
        gap={
          (boxOptions?.cssGap || cssGap || typeof gap === 'object') &&
          gap &&
          gap !== 'none' &&
          border !== 'between' &&
          border?.side !== 'between' &&
          (!Array.isArray(border) ||
            !border.find((b) => b.side === 'between')) &&
          gap
        }
        kindProp={kind}
        overflowProp={overflow}
        wrapProp={wrap}
        widthProp={width}
        heightProp={height}
        responsive={responsive}
        responsiveContainer={responsiveProp === 'container'}
        tabIndex={adjustedTabIndex}
        {...clickProps}
        {...passThemeFlag}
        {...rest}
        {...skeletonProps}
      >
        <ThemeContext.Provider value={nextTheme}>
          {contents}
        </ThemeContext.Provider>
      </StyledBox>
    );

    if (responsiveProp === 'container') {
      if (supportsContainerQueries()) {
        content = (
          <ResponsiveContext.Provider value={containerBreakpoint}>
            <ResponsiveContainerContext.Provider
              value={responsiveContainerValue}
            >
              {content}
            </ResponsiveContainerContext.Provider>
          </ResponsiveContext.Provider>
        );
      } else {
        console.warn(
          '<Box responsive="container"> requires styled-components v6 or later',
        );
      }
    }

    if (onClick) {
      content = <Keyboard onEnter={onClick}>{content}</Keyboard>;
    }

    return content;
  },
);

Box.displayName = 'Box';
Box.propTypes = BoxPropTypes;
export { Box };
