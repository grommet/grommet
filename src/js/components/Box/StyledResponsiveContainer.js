import styled from 'styled-components';
import { styledComponentsConfig } from '../../utils/styles';

export const StyledResponsiveContainer = styled.div.withConfig(
  styledComponentsConfig,
)`
  container-type: inline-size;
`;
