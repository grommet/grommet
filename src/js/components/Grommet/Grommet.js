import React, { forwardRef, useEffect, useMemo, useRef, useState } from 'react';
import { createGlobalStyle } from 'styled-components';

import {
  ContainerTargetContext,
  ResponsiveContext,
  ThemeContext,
} from '../../contexts';

import {
  deepMerge,
  backgroundIsDark,
  deviceResponsive,
  getBreakpoint,
  normalizeColor,
  useForwardedRef,
} from '../../utils';
import { base as baseTheme } from '../../themes';
import { StyledGrommet } from './StyledGrommet';
import { RootsContext } from '../../contexts/RootsContext';
import { OptionsContext } from '../../contexts/OptionsContext';
import { format, MessageContext } from '../../contexts/MessageContext';
import defaultMessages from '../../languages/default.json';
import { GrommetPropTypes } from './propTypes';
import { AnalyticsProvider } from '../../contexts/AnalyticsContext';

const FullGlobalStyle = createGlobalStyle`
  body { margin: 0; }
`;

const defaultOptions = {};

const Grommet = forwardRef((props, ref) => {
  const {
    children,
    full,
    containerTarget = typeof document === 'object' ? document.body : undefined,
    theme: themeProp,
    options = defaultOptions,
    messages: messagesProp,
    onAnalytics,
    ...rest
  } = props;
  const { background, dir, themeMode, userAgent } = props;
  const [stateResponsive, setResponsive] = useState();

  const theme = useMemo(() => {
    const nextTheme = deepMerge(baseTheme, themeProp || {});

    // if user provides specific menu alignment, we don't want
    // the defaults to be included at all (can cause issues with controlMirror)
    // override merged value with themeProp value
    if (
      themeProp &&
      themeProp.menu &&
      themeProp.menu.drop &&
      themeProp.menu.drop.align
    ) {
      delete nextTheme.menu.drop.align;
      nextTheme.menu.drop.align = themeProp.menu.drop.align;
    }
    const {
      colors: { background: themeBackground },
    } = nextTheme.global;

    nextTheme.dark = (themeMode || nextTheme.defaultMode) === 'dark';

    if (
      themeMode === 'auto' &&
      typeof window !== 'undefined' &&
      window.matchMedia &&
      window.matchMedia('(prefers-color-scheme: dark)').matches
    ) {
      nextTheme.dark = true;
    }

    const color = normalizeColor(background || themeBackground, nextTheme);
    nextTheme.dark = backgroundIsDark(color, nextTheme);
    nextTheme.baseBackground = background || themeBackground;
    // This allows DataTable to intelligently set the background of a pinned
    // header or footer.
    nextTheme.background = nextTheme.baseBackground;

    if (dir) {
      nextTheme.dir = dir;
    }

    return nextTheme;
  }, [background, dir, themeMode, themeProp]);

  const messages = useMemo(() => {
    // combine the passed in messages, if any, with the default
    // messages and format function.
    const nextMessages = deepMerge(
      defaultMessages,
      messagesProp?.messages || {},
    );
    return {
      messages: nextMessages,
      format: (opts) => {
        const message = messagesProp?.format && messagesProp.format(opts);
        return typeof message !== 'undefined'
          ? message
          : format(opts, nextMessages);
      },
    };
  }, [messagesProp]);

  useEffect(() => {
    const onResize = () => {
      setResponsive(getBreakpoint(document.body.clientWidth, theme));
    };
    window.addEventListener('resize', onResize);
    onResize();
    return () => {
      window.removeEventListener('resize', onResize);
    };
  }, [theme]);

  const responsive =
    stateResponsive ||
    deviceResponsive(userAgent, theme) ||
    theme.global.deviceBreakpoints.tablet;

  const grommetRef = useForwardedRef(ref);

  // track open FocusedContainers in a global array to manage
  // focus event listeners for trapFocus
  const roots = useRef([]);
  useEffect(() => {
    if (grommetRef.current) roots.current.push(grommetRef.current);
  }, [grommetRef]);
  const rootsContextValue = useMemo(() => ({ roots }), []);

  return (
    <ThemeContext.Provider value={theme}>
      <ResponsiveContext.Provider value={responsive}>
        <RootsContext.Provider value={rootsContextValue}>
          <ContainerTargetContext.Provider value={containerTarget}>
            <OptionsContext.Provider value={options}>
              <MessageContext.Provider value={messages}>
                <AnalyticsProvider onAnalytics={onAnalytics}>
                  <StyledGrommet full={full} {...rest} ref={grommetRef}>
                    {children}
                  </StyledGrommet>
                  {full && <FullGlobalStyle />}
                </AnalyticsProvider>
              </MessageContext.Provider>
            </OptionsContext.Provider>
          </ContainerTargetContext.Provider>
        </RootsContext.Provider>
      </ResponsiveContext.Provider>
    </ThemeContext.Provider>
  );
});

Grommet.displayName = 'Grommet';
Grommet.propTypes = GrommetPropTypes;

export { Grommet };
