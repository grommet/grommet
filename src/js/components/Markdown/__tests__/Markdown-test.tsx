import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { hpe } from 'grommet-theme-hpe';

import { Grommet } from '../../Grommet';
import { Markdown } from '..';
import { Box } from '../../Box';

const CONTENT = `
# H1

Paragraph

## H2

### H3

#### H4

[a link](#)

> i carry your heart with me

![alt text](//v2.grommet.io/assets/IMG_4245.jpg "Markdown Image")

Markdown | Less | Pretty
--- | --- | ---
*Still* | \`renders\` | **nicely**
1 | 2 | 3
`;

const Wrapper = (props: any) => <Box gap="small" {...props} />;

test('Markdown renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Markdown>{CONTENT}</Markdown>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Markdown renders without grommet wrapper', () => {
  const { container } = render(<Markdown>{CONTENT}</Markdown>);

  expect(container.firstChild).toMatchSnapshot();
});

test('wrapper', () => {
  const { container } = render(
    <Grommet>
      <Markdown options={{ wrapper: Wrapper }}>{CONTENT}</Markdown>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
