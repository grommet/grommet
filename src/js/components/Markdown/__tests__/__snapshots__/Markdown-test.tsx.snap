// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Markdown renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c3 {
  font-size: 1.75rem;
  line-height: 2rem;
  max-width: 816px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c4 {
  font-size: 1.5rem;
  line-height: 1.75rem;
  max-width: 624px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c5 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.5rem;
  max-width: 30em;
  font-weight: undefined;
}

.c9 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c6 {
  box-sizing: border-box;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  color: #006750;
  font-weight: 500;
  text-decoration: underline;
  cursor: pointer;
}

.c6:hover {
  text-decoration: underline;
  color: #074b3b;
}

.c8 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.36);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c8 >div {
  height: 100%;
  justify-content: center;
}

.c8 label {
  justify-content: center;
}

.c10 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.12);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
  padding-top: 12px;
  padding-bottom: 11px;
}

.c7 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

.c7 tbody tr:last-child td {
  border-color: transparent;
}

.c7 tbody tr:last-child th {
  border-color: transparent;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 1.5rem;
    line-height: 1.75rem;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 1.25rem;
    line-height: 1.5rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    font-size: 1.125rem;
    line-height: 1.25rem;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <div>
    <h1
      class="c1"
      id="h1"
    >
      H1
    </h1>
    <p
      class="c2"
    >
      Paragraph
    </p>
    <h2
      class="c3"
      id="h2"
    >
      H2
    </h2>
    <h3
      class="c4"
      id="h3"
    >
      H3
    </h3>
    <h4
      class="c5"
      id="h4"
    >
      H4
    </h4>
    <p
      class="c2"
    >
      <a
        class="c6"
        href="#"
      >
        a link
      </a>
    </p>
    <blockquote>
      <p
        class="c2"
      >
        i carry your heart with me
      </p>
    </blockquote>
    <p
      class="c2"
    >
      <img
        alt="alt text"
        class=""
        src="//v2.grommet.io/assets/IMG_4245.jpg"
        title="Markdown Image"
      />
    </p>
    <table
      class="c7"
    >
      <thead
        class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="c8"
          >
            <div
              class="c9"
            >
              Markdown
            </div>
          </td>
          <td
            class="c8"
          >
            <div
              class="c9"
            >
              Less
            </div>
          </td>
          <td
            class="c8"
          >
            <div
              class="c9"
            >
              Pretty
            </div>
          </td>
        </tr>
      </thead>
      <tbody
        class="StyledTable__StyledTableBody-sc-1m3u5g-3"
      >
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="c10"
          >
            <em>
              Still
            </em>
          </td>
          <td
            class="c10"
          >
            <code>
              renders
            </code>
          </td>
          <td
            class="c10"
          >
            <strong>
              nicely
            </strong>
          </td>
        </tr>
        <tr
          class="StyledTable__StyledTableRow-sc-1m3u5g-2"
        >
          <td
            class="c10"
          >
            1
          </td>
          <td
            class="c10"
          >
            2
          </td>
          <td
            class="c10"
          >
            3
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`Markdown renders without grommet wrapper 1`] = `
.c0 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c1 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

.c8 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c5 {
  box-sizing: border-box;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer;
}

.c5:hover {
  text-decoration: underline;
}

.c7 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c9 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c6 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

@media only screen and (max-width: 768px) {
  .c0 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div>
  <h1
    class="c0"
    id="h1"
  >
    H1
  </h1>
  <p
    class="c1"
  >
    Paragraph
  </p>
  <h2
    class="c2"
    id="h2"
  >
    H2
  </h2>
  <h3
    class="c3"
    id="h3"
  >
    H3
  </h3>
  <h4
    class="c4"
    id="h4"
  >
    H4
  </h4>
  <p
    class="c1"
  >
    <a
      class="c5"
      href="#"
    >
      a link
    </a>
  </p>
  <blockquote>
    <p
      class="c1"
    >
      i carry your heart with me
    </p>
  </blockquote>
  <p
    class="c1"
  >
    <img
      alt="alt text"
      class=""
      src="//v2.grommet.io/assets/IMG_4245.jpg"
      title="Markdown Image"
    />
  </p>
  <table
    class="c6"
  >
    <thead
      class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c7"
        >
          <div
            class="c8"
          >
            Markdown
          </div>
        </td>
        <td
          class="c7"
        >
          <div
            class="c8"
          >
            Less
          </div>
        </td>
        <td
          class="c7"
        >
          <div
            class="c8"
          >
            Pretty
          </div>
        </td>
      </tr>
    </thead>
    <tbody
      class="StyledTable__StyledTableBody-sc-1m3u5g-3"
    >
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c9"
        >
          <em>
            Still
          </em>
        </td>
        <td
          class="c9"
        >
          <code>
            renders
          </code>
        </td>
        <td
          class="c9"
        >
          <strong>
            nicely
          </strong>
        </td>
      </tr>
      <tr
        class="StyledTable__StyledTableRow-sc-1m3u5g-2"
      >
        <td
          class="c9"
        >
          1
        </td>
        <td
          class="c9"
        >
          2
        </td>
        <td
          class="c9"
        >
          3
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`wrapper 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c2 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c5 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c6 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c7 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
}

.c1 {
  display: flex;
  box-sizing: border-box;
  max-width: 100%;
  min-width: 0;
  min-height: 0;
  flex-direction: column;
}

.c3 {
  flex: 0 0 auto;
  align-self: stretch;
  height: 12px;
}

.c8 {
  box-sizing: border-box;
  display: inline-flex;
  font-size: inherit;
  line-height: inherit;
  color: #7D4CDB;
  font-weight: 600;
  text-decoration: none;
  cursor: pointer;
}

.c8:hover {
  text-decoration: underline;
}

.c10 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  border-bottom: solid 1px rgba(0, 0, 0, 0.33);
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c11 {
  margin: 0;
  padding: 0;
  font-weight: inherit;
  text-align: inherit;
  text-align: start;
  padding-left: 12px;
  padding-right: 12px;
  padding-top: 6px;
  padding-bottom: 6px;
}

.c9 {
  border-spacing: 0;
  border-collapse: collapse;
  width: inherit;
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c6 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c7 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    height: 6px;
  }
}

<div
  class="c0"
>
  <div>
    <div
      class="c1"
    >
      <h1
        class="c2"
        id="h1"
      >
        H1
      </h1>
      <div
        class="c3"
      />
      <p
        class="c4"
      >
        Paragraph
      </p>
      <div
        class="c3"
      />
      <h2
        class="c5"
        id="h2"
      >
        H2
      </h2>
      <div
        class="c3"
      />
      <h3
        class="c6"
        id="h3"
      >
        H3
      </h3>
      <div
        class="c3"
      />
      <h4
        class="c7"
        id="h4"
      >
        H4
      </h4>
      <div
        class="c3"
      />
      <p
        class="c4"
      >
        <a
          class="c8"
          href="#"
        >
          a link
        </a>
      </p>
      <div
        class="c3"
      />
      <blockquote>
        <p
          class="c4"
        >
          i carry your heart with me
        </p>
      </blockquote>
      <div
        class="c3"
      />
      <p
        class="c4"
      >
        <img
          alt="alt text"
          class=""
          src="//v2.grommet.io/assets/IMG_4245.jpg"
          title="Markdown Image"
        />
      </p>
      <div
        class="c3"
      />
      <table
        class="c9"
      >
        <thead
          class="StyledTable__StyledTableHeader-sc-1m3u5g-4"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="c10"
            >
              <div
                class="c1"
              >
                Markdown
              </div>
            </td>
            <td
              class="c10"
            >
              <div
                class="c1"
              >
                Less
              </div>
            </td>
            <td
              class="c10"
            >
              <div
                class="c1"
              >
                Pretty
              </div>
            </td>
          </tr>
        </thead>
        <tbody
          class="StyledTable__StyledTableBody-sc-1m3u5g-3"
        >
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="c11"
            >
              <em>
                Still
              </em>
            </td>
            <td
              class="c11"
            >
              <code>
                renders
              </code>
            </td>
            <td
              class="c11"
            >
              <strong>
                nicely
              </strong>
            </td>
          </tr>
          <tr
            class="StyledTable__StyledTableRow-sc-1m3u5g-2"
          >
            <td
              class="c11"
            >
              1
            </td>
            <td
              class="c11"
            >
              2
            </td>
            <td
              class="c11"
            >
              3
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;
